### PR TITLE
fix: notification indicator

### LINKED ISSUE #1655: Bug: Persistent Yellow Circle on Notification Bell After Viewing Notifications on OpenSauced
### Describe the bug

When using OpenSauced, I've noticed that a yellow circle appears on the notification bell to indicate new notifications. However, after I view these notifications, the yellow circle remains there, which can be confusing and misleading for users.

**Expected Behavior:**
The yellow circle should disappear from the notification bell icon once the user has viewed their notifications. This behavior is essential for a clear and intuitive user experience.

**Actual Behavior:**
The yellow circle persists even after viewing the notifications, causing confusion and making it difficult for users to know if they have new notifications or not.

**Additional Information:**
This issue affects the overall user experience on OpenSauced and could lead to missed notifications. It is recommended to resolve this issue to improve the platform's usability

Screenshot_20230901-222539_1

### Steps to reproduce

1. Log in to your OpenSauced account.
2. Click on the notification bell icon or access the notifications.
3. Observe that a yellow circle appears to indicate new notifications.
4. Click on the notifications to view them.
5. After viewing the notifications, the yellow circle remains on the notification bell icon.

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by RitaDee**
## Description

This PR addresses the persistent orange notification indicator circle. The changes ensure that the indicator is cleared when the button is clicked.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1655

## Mobile & Desktop Screenshots/Recordings

Before:
image

After:

image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: components/molecules/AuthSection/auth-section.tsx [hunk 1]
```diff
@@ -117,7 +117,16 @@ const AuthSection: React.FC = ({}) => {
                   <Divider type="vertical" className="!h-6 !bg-light-slate-6"></Divider>
                 </>
               )}
-              <button className="relative cursor-pointer" onClick={() => router.push(`/user/notifications`)}>
+              <button
+                key="notifications"
```
Comment (nickytonline) [inline components/molecules/AuthSection/auth-section.tsx:121]: Refactor (blocking): The `key` prop is only required for items in an array.

File: components/molecules/AuthSection/auth-section.tsx [hunk 2]
```diff
@@ -117,7 +117,16 @@ const AuthSection: React.FC = ({}) => {
...
+                className="relative cursor-pointer"
+                onClick={() => {
+                  if (userInfo && userInfo.notification_count > 0) {
+                    setUserInfo({ ...userInfo, notification_count: 0 });
```
Comment (nickytonline) [inline components/molecules/AuthSection/auth-section.tsx:125]: Does this still work if the page is refreshed? I imagine the user in the database needs to update the notification count?

**Review by nickytonline — CHANGES_REQUESTED** [review]

File: components/molecules/AuthSection/auth-section.tsx (hunk 2 above)
Comment (RitaDee) [reply components/molecules/AuthSection/auth-section.tsx:125]: I will give this a thought.

File: components/molecules/AuthSection/auth-section.tsx (hunk 1 above)
Comment (RitaDee) [reply components/molecules/AuthSection/auth-section.tsx]: Done.

File: components/molecules/AuthSection/auth-section.tsx (hunk 2 above)
Comment (RitaDee) [reply components/molecules/AuthSection/auth-section.tsx:124]: It appears to be updating when refreshed. 


https://github.com/open-sauced/app/assets/106407013/4ce1ce7b-3b79-4fcb-a12a-5ca0776f6c1d

Comment (nickytonline) [reply components/molecules/AuthSection/auth-section.tsx:124]: One last test for the fix. If you log out, close the browser then reopen everything and log in, does it still stay cleared?

Comment (RitaDee) [reply components/molecules/AuthSection/auth-section.tsx:124]: Yes, it does.


https://github.com/open-sauced/app/assets/106407013/7f7bd57e-e1df-4f84-8b32-7b5ab62fb507

**Review by nickytonline — APPROVED** [review]
Thanks for the PR @RitaDee!

**Review by OgDev-01 — APPROVED** [review]
🍕

**Pull request merged**